### PR TITLE
Fix the expiration by version logic

### PR DESCRIPTION
**Pull request by Dexterp37**
### Pull Request checklist ###

- [ ] **Quality**: This PR builds and tests run cleanly
  - `make test` runs without emitting any warnings
  - `make lint` runs without emitting any errors
- [ ] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [ ] **Changelog**: This PR includes a changelog entry to `CHANGELOG.md` or an explanation of why it does not need one
  - Any breaking changes to language binding APIs are noted explicitly

### CODE REVIEW

File: tests/test_parser.py [hunk 1]
```diff
@@ -641,15 +641,15 @@ def test_expire_by_major_version():
...
+                    "expires": 7,
                 },
                 "metric_expired": {
                     "type": "boolean",
                     "expires": "expired",
                 },
                 "metric": {
                     "type": "boolean",
-                    "expires": 7,
+                    "expires": 18,
```
Comment (badboy) [inline tests/test_parser.py:652]: Can you add one more metric that expires in v15?
Just so we have a test that's explicitly the edge case ("expires" defines the first version in which this metric is expired)

**Review by badboy — APPROVED** [review]

File: tests/test_parser.py (hunk 1 above)
Comment (Dexterp37) [reply tests/test_parser.py:656]: Sure, done!
